app/javascript/components/App.js

```diff
@@ -49,16 +52,75 @@ export default class App extends React.Component {
         />
 
         <Switch>
-          <Route exact path="/" component={Home}/>
-          <Route path="/library" component={Library}/>
-          <Route path="/borrow" component={Borrow}/>
-          <Route path="/borrowshow" component={BorrowShow}/>
-          <Route path="/borrowconfirmation" component={BorrowConfirmation}/>
-          <Route path="/borrowedshow" component={BorrowedShow}/>
-          <Route path="/lend" component={Lend}/>
-          <Route path="/lendconfirmation" component={LendConfirmation}/>
-          <Route path="/lendedshow" component={LendedShow}/>
-          <Route component={NotFound}/>
+
+          <Route exact path="/" component={Home} />
+
+          <Route path="/library"
+            render={(props) => {
+              let user = current_user.id
+              let books = this.state.books.filter(book => book.user_id === user)
+              return (
+                <MyLibrary
+                  {...props}
+                  books={books}
+                />
+              )
+            }}
+          />
+
+          <Route exact path="/borrow"
+            render={(props) => {
+              return (
+                <Borrow
+                  books={this.state.books}
+                />
+              )
+            }}
+          />
+
+          <Route
+            path="/borrow/:id"
+            render={(props) => {
+              let id = props.match.params.id
+              let book = this.state.books.find(book => book.id === parseInt(id))
```

Comment:
If you are doing just a simple string to int conversion, you can also use `Number(id)`.  Up to you!